### PR TITLE
MSC2912: Setting cross-signing keys during registration

### PR DESCRIPTION
**Pull request by uhoreg**
[Rendered](https://github.com/uhoreg/matrix-doc/blob/registration_cross-signing/proposals/2912-setting-cross-signing-keys-during-registration.md)

### CODE REVIEW

File: proposals/2912-setting-cross-signing-keys-during-registration.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# MSC2912: Setting cross-signing keys during registration
+
+Setting cross-signing keys (the `/keys/device_signing/upload` endpoint)
+requires UI Authentication as an extra layer of security to ensure that an
+attacker cannot easily replace a user's cross-signing keys with one that they
+control.  One of the side-effects of this is that when a user registers an
+account, the client will need to another UI Auth to upload the user's
+cross-signing key.  Some clients temporarily cache the user's password when
+registering, to avoid prompting the user a second time for their password.
```
Comment (clokep) [inline proposals/2912-setting-cross-signing-keys-during-registration.md:9]: Having the client cache the password and automatically provide it seems like something that they shouldn't be doing anyway 👎

File: proposals/2912-setting-cross-signing-keys-during-registration.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+However, this does not work for users that sign in using Single Sign-on.
```
Comment (deepbluev7) [inline proposals/2912-setting-cross-signing-keys-during-registration.md:10]: Doesn't SSO use /login, without ever calling /register? Am I missing something?

Comment (clokep) [reply proposals/2912-setting-cross-signing-keys-during-registration.md:10]: I had the same thought last evening. 👍

**Comment by uhoreg** [comment]
Since this doesn't even solve the SSO problem, I'm withdrawing this.

**Pull request closed without merging**